plugins/shell/src/process/mod.rs

```diff
@@ -364,11 +375,32 @@ fn spawn_pipe_reader<F: Fn(Vec<u8>) -> CommandEvent + Send + Copy + 'static>(
     guard: Arc<RwLock<()>>,
     pipe_reader: PipeReader,
     wrapper: F,
+    raw_out: bool
 ) {
     spawn(move || {
         let _lock = guard.read().unwrap();
         let mut reader = BufReader::new(pipe_reader);
 
+        if raw_out {
+            loop {
+                let result = reader.fill_buf();
+                match result {
+                    Ok(buf) => {
+                        let length = buf.len();
+                        if length == 0 { break; }
+                        let tx_ = tx.clone();
+                        let _ = block_on_task(async move { tx_.send(wrapper(buf.to_vec())).await });
+                        reader.consume(length);
+                    }
+                    Err(e) => {
+                        let tx_ = tx.clone();
+                        let _ = block_on_task( async move { tx_.send(CommandEvent::Error(e.to_string())).await });
+                    }
+                }
+            }
+            return;
+        } 
```

Comment:
lets move the current logic and the new logic into two separate function and then call them conditionally:

```rs
if raw {
  read_raw_bytes(&mut reader)
} else {
  read_line(&mut reader)
}
```

Reply:
Hopefully implemented the way you expected!